x-pack/plugins/enterprise_search/public/applications/shared/hidden_text/hidden_text.tsx

```diff
@@ -0,0 +1,38 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import React, { useState, ReactElement } from 'react';
+import { i18n } from '@kbn/i18n';
+
+interface ChildrenProps {
+  toggle: () => void;
+  isHidden: boolean;
+  hiddenText: React.ReactNode;
+}
+
+interface Props {
+  text: string;
+  children(props: ChildrenProps): ReactElement;
+}
+
+export const HiddenText: React.FC<Props> = ({ text, children }) => {
```

Comment:
I don't super care about this personally, but if you want to do the I prefixing thing mentioned elsewhere in this PR

```suggestion
interface IChildrenProps {
  toggle: () => void;
  isHidden: boolean;
  hiddenText: React.ReactNode;
}

interface IProps {
  text: string;
  children(props: IChildrenProps): ReactElement;
}

export const HiddenText: React.FC<IProps> = ({ text, children }) => {
```